Model-view pattern drafted, minor fix

topminime opened the pull request:
#3 and #4 to be closed.
note: CModel implements singleton pattern

topminime committed:
todoList

topminime committed:
Merge branch 'master' of https://github.com/serzap/todoListApp.git into minime

# Conflicts:
#	TodoListApp.pro.user

topminime committed:
Refactoring + minor fixes
Code refactored. Model-view pattern implemented using slot-signal mechanism.
fixed minor problem task name is now deleted from window after being added to the list

serzap reviewed (commented):
Please don't push this file TodoListApp.pro.user. Git ignore should prevent pushing starting from next commit.

serzap commented on CModel.cpp:
```diff
@@ -0,0 +1,21 @@
+
+CModel *CModel::getInstance()
+{
+    if (!theModel) theModel = new CModel();
```
Memory leak. Pls delete allocated memory.

serzap commented on CTask.cpp:
```diff
@@ -0,0 +1,33 @@
+    return mDescription;
+}
+
+void CTask::setDescription(const std::string &description)
```
Better use QString

serzap commented on CTask.hpp:
```diff
@@ -0,0 +1,25 @@
+{
+public:
+    CTask();
+    CTask(std::string description);
```
QString

serzap commented on CTask.hpp:
```diff
@@ -0,0 +1,25 @@
+    bool getStatus() const;
+    void setStatus(bool status);
+
+    std::string getDescription() const;
```
QString

serzap commented on MainWindow.hpp:
```diff
@@ -19,13 +20,15 @@ class MainWindow : public QMainWindow
+
 signals:
-    void newTaskButtonClicked();
+    void newTaskButtonClicked(std::string taskName);
```
const QString&

serzap commented on NewTaskWindow.hpp:
```diff
@@ -17,14 +17,14 @@ class NewTaskWindow : public QWidget
 
 signals:
-    void newTaskCreated(TaskInfo taskInfo);
+    void newTaskCreated(CTask task);
```
const CTask&

serzap reviewed (commented):
See comments

topminime committed:
MVC draft v2.0, renaming classes
MVC remade and redrafted
few classes renamed (CModel -> TaskController), added Application class

topminime committed:
minor fixes
fixed memory leaks
string replaced with QString
arguments passed by constant references where needed

topminime committed:
taskList autorefresh every second
if a task is overdue it is re-painted red

topminime committed:
CurrentTaskWindow added + delete/back button

topminime committed:
Revert "CurrentTaskWindow added + delete/back button"
This reverts commit 0a20587c81dbd997069e1795b243aff107def301.

topminime committed:
Revert "CurrentTaskWindow added + delete/back button"

This reverts commit 0a20587c81dbd997069e1795b243aff107def301.

 Changes to be committed:

serzap committed:
Merge branch 'minime' of github.com:serzap/todoListApp into minime

serzap committed:
Refactoring.
- Added const
- Removed dynamic alloc
- Changed Singleton implementation
- Added taskID

serzap commented on MainWindow.cpp:
```diff
@@ -17,20 +18,25 @@ void MainWindow::on_pushButton_NewTask_clicked()
-    for (auto task = list.begin(); task != list.end(); task++) {
-        mUI->listWidget_Tasks->addItem(getListItem(*task));
+    QString itemInfo = QString(task.getDescription()+'\t' + task.getDeadline().toString("dd.MM.yyyy hh:mm"));
+    QListWidgetItem* item = new QListWidgetItem(itemInfo);
```
Memory leak. Implemetation should be changed.

serzap committed:
Logger added.
